lib/charms/mongodb/v0/mongodb_secrets.py

```diff
@@ -0,0 +1,125 @@
+"""Secrets related helper classes/functions."""
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from typing import Dict, Optional
+
+from ops import Secret, SecretInfo
+from ops.charm import CharmBase
+from ops.model import SecretNotFoundError
+
+from config import Config
+from exceptions import SecretAlreadyExistsError
+
+# The unique Charmhub library identifier, never change it
+
+# FIXME: Incorrect libid, as my charmcraft authentication is missing 'package access' :-/
+LIBID = "b9a7fe0c38d8486a9d1ce94c27d4758e"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 8
+
+APP_SCOPE = Config.Relations.APP_SCOPE
+UNIT_SCOPE = Config.Relations.UNIT_SCOPE
+Scopes = Config.Relations.Scopes
+
+
+def generate_secret_label(charm: CharmBase, scope: Scopes) -> str:
+    """Generate unique group_mappings for secrets within a relation context."""
+    members = [charm.app.name, scope]
+    return f"{'.'.join(members)}"
+
```

Comment:
I really think this function should be left separate from the `SecretCache` class(es).

**Rationale:**

`SecretCache`: Providing an abstraction layer above direct Juju access WITH caching. I.e. making sure not to fetch a secret multiple times. (*)
Now, if at some point we decide we want caching no more, this is the single, encapsulated logic that is to be changed (dropped). `SecretChache` and `CachedSecret` are to provide this level of isolation.

`generate_secret_label`: The logic how to define it's secret labesl **belongs ot the application**. Each charm can decide how to label their secrets. This is a different separation of concern either from Juju access or from internal implementation details whether we're cacnihg secrets or not. This is part of the Application Logic.

With all the above in mind, pls in case someone changing the PR -- if possible pls keep `generate_secrets_label()` separate from the abstract `Secret`-manipulation classes. This function is outside of that scope.

(*) For multiple reasons actullay, including that the secret may have changed. If we're not carefully passing around our already-fetched secret (when dealing with a chain of called functions), but fetch it again, we should get the latest version... Which could be different from what we had at the start of the event handler ;-)

Reply:
Yes, this info in the doc strings would be awesome! Thank you so much for taking time to add all this info.